spec/ruby/shared/fiber/resume.rb

```diff
@@ -36,16 +36,26 @@
   end
 
   it "runs until Fiber.yield" do
-    obj = mock('obj')
-    obj.should_not_receive(:do)
-    fiber = Fiber.new { 1 + 2; Fiber.yield; obj.do }
-    fiber.send(@method)
+    if @method == :transfer
+      fiber = Fiber.new { 1 + 2; Fiber.yield; }
+      -> { fiber.send(@method) }.should raise_error(FiberError)
+    else
+      obj = mock('obj')
+      obj.should_not_receive(:do)
+      fiber = Fiber.new { 1 + 2; Fiber.yield; obj.do }
+      fiber.send(@method)
+    end
```

Comment:
It would be better to move these specs to `spec/ruby/core/fiber/resume_spec.rb` and `spec/ruby/core/fiber/transfer_spec.rb` since the behavior now differs between them, could you do that?

Reply:
ok. and we need version guard...

Reply:
Yes, for behavior that changes, we need `ruby_version_is` guards